src/scripts/main.js

```diff
@@ -1,3 +1,319 @@
 'use strict';
 
-// write your code here
+const controls = document.querySelector('.controls');
+const scoreInfo = document.querySelector('.game-score');
+
+const rows = document.querySelectorAll('.field-row');
+
+const msgStart = document.querySelector('.message-start');
+const msgWin = document.querySelector('.message-win');
+const msgLose = document.querySelector('.message-lose');
+
+controls.insertAdjacentHTML('beforeend', `
+  <button
+    class="button restart hidden"
+    >Restart
+  </button>
+`);
+
+const startBtn = document.querySelector('.start');
+const restartBtn = document.querySelector('.restart');
+
+const game = {
+  isRunning: false,
+  isWon: false,
+  isLost: false,
+  score: 0,
+  rowsQty: rows.length,
+  colsQty: rows[0].children.length,
+  matrix: [],
+};
+
+const moves = {
+  moveRight(matrix) {
+    matrix.forEach(row => {
+      row.forEach((num, index) => {
+        if (num === 0) {
+          row.splice(index, 1);
+          row.unshift(0);
+        }
+      });
+
+      for (let i = row.length - 1; i >= 0; i--) {
+        if (row[i] > 0 && row[i] === row[i - 1]) {
```

Comment:
you repeat row[i] a lot of times
Would be better if you move this to variables
Also, you can use it in all cases